tests/components/idasen_desk/test_init.py

```diff
@@ -53,6 +54,77 @@ async def test_no_ble_device(hass: HomeAssistant, mock_desk_api: MagicMock) -> N
         assert entry.state is ConfigEntryState.SETUP_RETRY
 
 
+async def test_reconnect_on_bluetooth_callback(
+    hass: HomeAssistant, mock_desk_api: MagicMock
+) -> None:
+    """Test that a reconnet is made after the bluetooth callback is triggered."""
+    with mock.patch(
+        "homeassistant.components.idasen_desk.bluetooth.async_register_callback"
+    ) as mock_register_callback:
+        await init_integration(hass)
+        assert len(hass.config_entries.async_entries(DOMAIN)) == 1
+        mock_desk_api.connect.assert_called_once()
+        mock_register_callback.assert_called_once()
+
+        mock_desk_api.is_connected = False
+        _, register_callback_args, _ = mock_register_callback.mock_calls[0]
+        bt_callback = register_callback_args[1]
+        bt_callback(None, None)
+        await hass.async_block_till_done()
+        assert mock_desk_api.connect.call_count == 2
+
+
+async def test_duplicated_disconnect_is_no_op(
+    hass: HomeAssistant, mock_desk_api: MagicMock
+) -> None:
+    """Test that calling disconnect while disconnectig is a no-op."""
+    await init_integration(hass)
+
+    await hass.services.async_call(
+        "button", "press", {"entity_id": "button.test_disconnect"}, blocking=True
+    )
+    await hass.async_block_till_done()
+
+    async def mock_disconnect():
+        await asyncio.sleep(0.1)
```

Comment:
```suggestion
        await asyncio.sleep(0)
```

If you only want to yield to the event loop here, sleep 0 is implemented as a yield